Python/codegen.c

```diff
@@ -720,23 +727,75 @@ codegen_leave_annotations_scope(compiler *c, location loc,
     Py_SETREF(co->co_localsplusnames, new_names);
 
     _PyCompile_ExitScope(c);
-    if (co == NULL) {
-        return ERROR;
-    }
     int ret = codegen_make_closure(c, loc, co, 0);
     Py_DECREF(co);
     RETURN_IF_ERROR(ret);
     return SUCCESS;
 }
 
+static int
+codegen_deferred_annotations_body(compiler *c, location loc,
+    PyObject *deferred_anno, PyObject *conditional_annotation_indices, int scope_type)
+{
+    Py_ssize_t annotations_len = PyList_Size(deferred_anno);
+
+    assert(PyList_CheckExact(conditional_annotation_indices));
+    assert(annotations_len == PyList_Size(conditional_annotation_indices));
+
+    ADDOP_I(c, loc, BUILD_MAP, 0); // stack now contains <annos>
+
+    for (Py_ssize_t i = 0; i < annotations_len; i++) {
+        PyObject *ptr = PyList_GET_ITEM(deferred_anno, i);
+        stmt_ty st = (stmt_ty)PyLong_AsVoidPtr(ptr);
+        if (st == NULL) {
+            return ERROR;
+        }
+        PyObject *mangled = _PyCompile_Mangle(c, st->v.AnnAssign.target->v.Name.id);
+        if (!mangled) {
+            return ERROR;
+        }
+        PyObject *cond_index = PyList_GET_ITEM(conditional_annotation_indices, i);
+        assert(PyLong_CheckExact(cond_index));
+        long idx = PyLong_AsLong(cond_index);
```

Comment:
```suggestion
        long idx = PyLong_AS_LONG(cond_index);
```

Currently, it's the same as `PyLong_AsLong` but `PyLong_AS_LONG`'s comment says "It may be useful in the future". Some other places use this macro, but I don't know if it's legacy code or not.